picard_collectmultiplemetric_2.8.1/picard_collectmultiplemetrics_2-8-1.cwl

```diff
@@ -0,0 +1,215 @@
+      prefix: DB_SNP=
+      separate: false
+    doc: >-
+      VCF format dbSNP file, used to exclude regions around known polymorphisms
+      from analysis by some PROGRAMs; PROGRAMs whose CLP doesn't allow for this
+      argument will quietly ignore it. Default value: null.
+  - id: include_unpaired
+    type: boolean?
+    inputBinding:
+      position: 0
+      prefix: INCLUDE_UNPAIRED=true
+      separate: false
+    doc: >-
+      Include unpaired reads in CollectSequencingArtifactMetrics. If set to true
+      then all paired reads will be included as well - MINIMUM_INSERT_SIZE and
+      MAXIMUM_INSERT_SIZE will be ignored in CollectSequencingArtifactMetrics.
+      Default value: false. This option can be set to 'null' to clear the
+      default value. Possible values: {true, false}
+outputs:
+  - id: all_metrics
+    type: Directory
+    outputBinding:
+      glob: .
+      outputEval: |-
+        ${
+            if(inputs.output_file_name){
+                self[0].basename = inputs.output_file_name + '_picard_metrics';
+          return self[0]
+            } else {
+                self[0].basename =  'picard_metrics';
+          return self[0]
+            }
+        }
+label: picard_collectmultiplemetrices_2.8.1
+arguments:
+  - position: 0
+    prefix: ''
+    valueFrom: "${\n  if(inputs.memory_per_job && inputs.memory_overhead) {\n   \n    if(inputs.memory_per_job % 1000 == 0) {\n    \t\n      return \"-Xmx\" + (inputs.memory_per_job/1000).toString() + \"G\"\n    }\n    else {\n      \n      return \"-Xmx\" + Math.floor((inputs.memory_per_job/1000)).toString() + \"G\" \n    }\n  }\n  else if (inputs.memory_per_job && !inputs.memory_overhead){\n    \n    if(inputs.memory_per_job % 1000 == 0) {\n    \t\n      return \"-Xmx\" + (inputs.memory_per_job/1000).toString() + \"G\"\n    }\n    else {\n      \n      return \"-Xmx\" + Math.floor((inputs.memory_per_job/1000)).toString() + \"G\" \n    }\n  }\n  else if(!inputs.memory_per_job && inputs.memory_overhead){\n    \n    return \"-Xmx15G\"\n  }\n  else {\n    \n  \treturn \"-Xmx15G\"\n  }\n}"
+  - position: 0
+    prefix: '-jar'
+    valueFrom: /usr/local/bin/picard.jar
+  - position: 0
+    prefix: ''
+    valueFrom: CollectMultipleMetrics
+  - position: 0
+    prefix: O=
+    separate: false
+    valueFrom: |-
+      ${
+          if(inputs.output_file_name){
+              return inputs.output_file_name
+          } else {
+              return inputs.input.basename.replace(/.bam/,'_md.bam')
+          }
+      }
+requirements:
+  - class: ResourceRequirement
+    ramMin: "${\r  if(inputs.memory_per_job && inputs.memory_overhead) {\r   \r    return inputs.memory_per_job + inputs.memory_overhead\r  }\r  else if (inputs.memory_per_job && !inputs.memory_overhead){\r    \r   \treturn inputs.memory_per_job + 2000\r  }\r  else if(!inputs.memory_per_job && inputs.memory_overhead){\r    \r    return 15000 + inputs.memory_overhead\r  }\r  else {\r    \r  \treturn 17000 \r  }\r}"
```

Comment:
Need to comment ramMin and coreMin that have expression and add static values to it